src/tbb/governor.cpp

```diff
@@ -180,15 +180,14 @@ static std::uintptr_t get_stack_base(std::size_t stack_size) {
         pthread_attr_destroy(&np_attr_stack);
     }
 #endif /* __linux__ */
-    std::uintptr_t stack_base{};
     if (stack_limit) {
+        stack_size = (np_stack_size > 0) ? np_stack_size : fallback_stack_size;
```

Comment:
Since `stack_size` becomes sort of a return value of this function, it should always be assigned as part of the function execution disregarding from on which platform the function runs.
So, at the very minimum I would recommend assigning `stack_size` to `fallback_stack_size` value initially (perhaps, at the top of the `get_stack_attributes` function). Then consider removing all the `suppress_unused_warning` calls as it seems to be not needed anymore.

Furthermore, `np_stack_size` is only needed for `#if __linux__ && !__bg__` case, so we may move declaration inside that macro block. Thus, check for `np_stack_size > 0` and re-assigning `stack_size` to that value can be moved into the inner `if` as it is the only place where it can have useful value.

Reply:
Thanks for your kindly review. I updated the patch.